rolling-update-kubernetes/src/main/scala/akka/rollingupdate/kubernetes/PodDeletionCostAnnotator.scala

```diff
@@ -0,0 +1,221 @@
+      log.debug("Received MemberRemoved {}", m)
+      updateIfNewCost(deletionCost, membersByAgeDesc - m)
+
+    case PodAnnotated =>
+      log.debug("Annotation updated successfully to {}", deletionCost)
+      // cancelling an eventual retry in case the annotation succeeded in the meantime
+      timers.cancel(RetryTimerId)
+
+    case ScheduleRetry(ex) =>
+      log.error(s"Failed to update annotation: [$ex]. Retrying with fixed delay of {}.", costSettings.retryDelay)
+      timers.startSingleTimer(RetryTimerId, RetryAnnotate, costSettings.retryDelay)
+      context.become(underRetryBackoff(deletionCost, membersByAgeDesc))
+
+    case GiveUp(er: String) =>
+      log.error(
+        "There was a client error when trying to set pod-deletion-cost annotation. " +
+        "Not retrying, check configuration. Error: {}",
+        er)
+
+    case es => log.debug("Ignoring message {}", es)
+  }
+
+  private def underRetryBackoff(deletionCost: Int, membersByAgeDesc: SortedSet[Member]): Receive = {
+    case cs@ClusterEvent.CurrentClusterState(members, _, _, _, _) =>
+      log.debug("Received while on retry backoff CurrentClusterState {}", cs)
+      context.become(underRetryBackoff(deletionCost, membersByAgeDesc ++ members)) // ordering used is from the first operand (so, by age)
+
+    case ClusterEvent.MemberUp(m) =>
+      log.debug("Received while on retry backoff MemberUp {}", m)
+      context.become(underRetryBackoff(deletionCost, membersByAgeDesc + m))
+
+    case ClusterEvent.MemberRemoved(m, _) =>
+      log.debug("Received while on retry backoff MemberRemoved {}", m)
+      context.become(underRetryBackoff(deletionCost, membersByAgeDesc - m))
+
+    case RetryAnnotate =>
+      updateIfNewCost(deletionCost, membersByAgeDesc)
+
+    case es => log.debug("Ignoring message {}", es)
+  }
+
+  private def costRequestToResult(newCost: Int): Future[RequestResult] = {
+    log.debug(
+      "Updating pod-deletion-cost annotation for pod: [{}] with cost: [{}]. Namespace: [{}]",
+      settings.podName,
+      newCost,
+      podNamespace
+    )
+    val request = ApiRequests.podDeletionCost(settings, apiToken, podNamespace, newCost)
+    val response =
+      clientSslContext
+        .map(http.singleRequest(request, _))
+        .getOrElse(http.singleRequest(request))
+
+    toResult(response)
+  }
+
+  private def updateIfNewCost(existingCost: Int, membersByAgeDesc: immutable.SortedSet[Member]): Unit = {
+
+    val podsToAnnotate = membersByAgeDesc.take(costSettings.annotatedPodsNr)
+    val newCost: Int = OlderCostsMore.costOf(cluster.selfMember, podsToAnnotate).getOrElse(0)
+    log.debug(
+      "Calculated cost={} (previously {}) for member={} in members by age (desc): {}",
+      newCost,
+      existingCost,
+      cluster.selfMember,
+      membersByAgeDesc)
+
+    if (newCost != existingCost) {
+      costRequestToResult(newCost).pipeTo(self)
+      context.become(onMessage(newCost, membersByAgeDesc))
```

Comment:
I haven't checked in detail. Does it still handle several subsequent failures and continue retrying until successful? Thinking about newCost vs existingCost for the failure retry scenario